src/decoder/mod.rs

```diff
@@ -895,6 +898,18 @@ impl<R: Read + Seek> Decoder<R> {
         self.get_tag(tag)?.into_string()
     }
 
+    pub fn tag_iter(&mut self) -> impl Iterator<Item = TiffResult<(Tag, ifd::Value)>> + '_ {
+        match self.image().ifd.as_ref() {
+            None => Either::Left(std::iter::empty()),
+            Some(ifd) => Either::Right(TagIter::new(
```

Comment:
Rather than constructing a TagIter here, I think we could directly return:
```rust
ifd.clone().into_iter().map(...)
```

Reply:
I tried but failed to make the borrow checker happy with this approach. Happy to change it if you can provide a working solution.

Reply:
I was able to get it to work with:

```rust
        self.image.ifd.as_ref().unwrap().iter().map(|(tag, entry)| {
            entry
                .val(&self.limits, self.bigtiff, &mut self.reader)
                .map(|value| (*tag, value))
        })
```

(You have to directly access the `self.image` field rather than calling the `self.image()` method to make the borrow check happy. There was a reason why the two are separate, though I don't remember what it is right now)

Reply:
Ah, cool, thx! I incorporated your changes and pushed the branch.